From 4e5d8e849a5a99dd9c57822b1a65ead7c14001d9 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Fri, 31 May 2024 16:41:48 +0200
Subject: [PATCH 0966/1695] plat-stm32mp2: add stm32mp21 compatibles in power
 drivers

In order to anticipate future issue, we prefer to create new compatibles
in power drivers for stm32mp21 as we cannot differ stm32mp25 and
stm32mp21 by reading registers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6d3f631280c5258117425f87d9283dc622b6d0a7 plat-stm32mp2: add stm32mp21 compatibles in power drivers]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I6083d2f59a80f23110e02b997615c222871ce0dc
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/384522
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c       | 1 +
 core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c   | 1 +
 core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c | 1 +
 3 files changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c
index 9e5dff584..e757696fc 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c
@@ -321,6 +321,7 @@ static TEE_Result stm32mp_pwr_probe(const void *fdt, int node,
 }
 
 static const struct dt_device_match stm32mp_pwr_match_table[] = {
+	{ .compatible = "st,stm32mp21-pwr" },
 	{ .compatible = "st,stm32mp25-pwr" },
 	{ }
 };
diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c
index cb7b16339..285cdbfe9 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_irq.c
@@ -466,6 +466,7 @@ stm32mp25_pwr_irq_user_dt_probe(const void *fdt, int node,
 }
 
 static const struct dt_device_match pwr_irq_test_match_table[] = {
+	{ .compatible = "st,stm32mp21-pwr-irq-user" },
 	{ .compatible = "st,stm32mp25-pwr-irq-user" },
 	{ }
 };
diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
index df9459f18..3b2214c04 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
@@ -584,6 +584,7 @@ static TEE_Result pwr_regulator_probe(const void *fdt, int node,
 }
 
 static const struct dt_device_match pwr_regulator_match_table[] = {
+	{ .compatible = "st,stm32mp21-pwr-regu" },
 	{ .compatible = "st,stm32mp25-pwr-regu" },
 	{ }
 };
-- 
2.47.2

